legion/engine/llri-dx/detail/instance.cpp

```diff
@@ -282,6 +280,10 @@ namespace LLRI_NAMESPACE
                 r = dx12Device->CreateCommandQueue(&dx12QueueDesc, IID_PPV_ARGS(&dx12Queue));
                 if (FAILED(r))
                 {
+                    // the internal queue resources for this queue havent been added to the device yet so they must be destroyed manually
+                    for (void* q : queues) { if (q) static_cast<ID3D12CommandQueue*>(q)->Release(); };
```

Comment:
have you tried compiling this?  idk for certain whether you can static_cast void ptrs?

Reply:
Yes this absolutely works

Reply:
also since values in containers could prolly change at some point, and you don't necessarily care about whatever is in them, i tend to use auto for iteration like this

Reply:
I chose to do so because it clarifies what the static_cast is casting from